### ISSUE TITLE
terragrunt file is selecting docker-bake .hcl

### ISSUE THREAD

**Issue by StefanSchoof** [super-linter/super-linter#2537]
## Describe the bug

Docker bake is using a file called docker-bake.hcl, see https://docs.docker.com/engine/reference/commandline/buildx_bake/#file

This is a hcl file, but not a file that terragrunt should check.

## Expected behavior

Having a valid docker-bake.hcl does not give any errors in super-linter

## Steps to Reproduce

1. Create a valid docker-bake.hcl like https://github.com/StefanSchoof/espresso/blob/main/docker-bake.hcl
2. Run super lint
3. Gets an error about this file from terragrunt

## Logs

```
2022-02-18 16:31:47 [ERROR]   Found errors in [terragrunt] linter!
2022-02-18 16:31:47 [ERROR]   Error code: 1. Command output:
------
time=2022-02-18T16:31:47Z level=error msg=Invalid file format /github/workspace/docker-bake.hcl
time=2022-02-18T16:31:47Z level=error msg=Unable to determine underlying exit code, so Terragrunt will exit with error code 1
```

## Additional context

#1707

**Comment by ferrarimarco** [comment]
Hi @StefanSchoof . Can you run the linter locally (as instructed by the bug template) with debug output turned on? We're interested in the part that populates the list of files to lint with terragrunt.

**Comment by StefanSchoof** [comment]
Hi @ferrarimarco, thanks for looking into it. Sorry that I missed that info in the initial post.

Here is the log: https://gist.github.com/StefanSchoof/bf6b67fd1bed0cfbaa65a4410eee65b7

A branch that is causing this error is https://github.com/StefanSchoof/espresso/tree/feature/superlint-terragrunt and run in the github action you can see at https://github.com/StefanSchoof/espresso/runs/5582975365.

**Comment by ferrarimarco** [comment]
I think this is, in fact, an issue. To "detect" a Terragrunt file, we just check the extension: https://github.com/github/super-linter/blob/e277edca06d0850200869e6552a94da09f95cb3a/lib/functions/buildFileList.sh#L744

IIRC, `.hcl` may not be necessarily specific to Terragrunt (or Terraform). To differentiate more and avoid false positives, we need to know more about Terragrunt files, [as we're doing for other file types](https://github.com/github/super-linter/blob/main/lib/functions/detectFiles.sh).

**Comment by ferrarimarco** [comment]
The lowest hanging fruit would be to exclude `docker-bake.hcl` as we did in https://github.com/github/super-linter/pull/1707, but maybe there's a more scalable approach?

**Comment by StefanSchoof** [comment]
Maybe the question, if terragrunt should handle hcl files at all should be ask. I personally never encountered an file hcl in the context of terraform. There all files have the tf extension.

**Comment by ferrarimarco** [comment]
The main Terragrunt configuration file is (by Terragrunt default) named: `terragrunt.hcl`. From the Terragrunt docs, it appears it might also support configuration using multiple HCL files, not just `terragrunt.hcl`, but I'm not sure. If it was only `terragrunt.hcl`, we could easily limit super-linter to that when building the list of files to lint.

**Comment by eitsupi** [comment]
Not directly related to this, but currently hadolint targets all files with "dockerfile" in the filename, so for example something like "dockerfile.md" will cause an error.
Would it be possible to add a setting that would allow users to exclude certain files, as is done in MegaLinter?
https://megalinter.github.io/latest/descriptors/terraform_terragrunt/
https://megalinter.github.io/latest/descriptors/dockerfile_hadolint/

**Comment by torvitas** [comment]
Hi everyone,
I am having a similar issue regarding terragrunt-lint and hcl files. In my case it is about vault policy configurations which are as well written as hcl and therefor managed in hcl files. As these are arbitrary named hcl files this problem cannot be fixed by just excluding a specific set of files. So a setting allowing me to exclude specific files or rather a glob fromt being terragrunted would be appreciated. On the other hand it probably would make more sense if terragrunt would not be applied to every hcl file by default as hcl is used in a lot of projects besides terraform.

**Comment by ferrarimarco** [comment]
I get your point. Selecting Terragrunt files based on the extension only might not be enough. Also, I didn't find a way to configure Terragrunt to ignore some files, so that's not an option, either.

The temporary workaround would be to disable Terragrunt by setting `VALIDATE_TERRAGRUNT` to `false`. That would still allow all the other linters to run while we think about a solution.

**Comment by torvitas** [comment]
That's exactly how we are working around it right now. Sadly we already had to disable terrascan due to the current symlink issue. There are not too many terraform linters left by now :-/

**Comment by ferrarimarco** [comment]
After checking the [terragrunt docs](https://terragrunt.gruntwork.io/docs/reference/cli-options/#hclfmt), it appears that terragrunt supports recursively searching for configuration files. We might use this feature instead of building the search logic ourselves.

**Comment by ferrarimarco** [comment]
Hi @torvitas!

We can cover the simple cases first, and eventually refine our approach later if there is a need.

#5178 should take care of this.
